README.md
Calendar_XH – a CMSimple_XH plugin
==================================

[![stable: 1.2.10](https://img.shields.io/badge/stable-1.2.10-green.svg)](https://github.com/cmb69/calendar_xh/releases/tag/1.2.10)
[![License: GPL v3](https://img.shields.io/badge/License-GPL%20v3-blue.svg)](http://www.gnu.org/licenses/gpl-3.0)

Copyright 2005-2006 Michael Svarrer  
Copyright 2007-2008 Tory  
Copyright 2008      Patrick Varlet  
Copyright 2011      Holger Irmler  
Copyright 2011-2013 Frank Ziesing  
Copyright 2017      Christoph M. Becker

Calendar_XH is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

Calendar_XH is distributed in the hope that it will be useful,
but *without any warranty*; without even the implied warranty of
*merchantability* or *fitness for a particular purpose*.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with Calendar_XH (see LICENSE).

For further information see the user manual in help/help.htm.

Christoph M. Becker  
Rheinstraße 147  
55424 Münster-Sarmsheim  
GERMANY  
mailto:cmbecker69@gmx.de


CHANGELOG
Version 1.2.10
* Update eliminating rare problems with magic quotes. Fix by cmb.

Version 1.2.9
* Update with ideas of cmb (empty lines bug on some servers, rel_"nofollow" added)

Version 1.2.8
* Security update (on adive by cmb)

Version 1.2.7 (10/2011)
* Slovak translation completed by Stanka
* Fixed bug in calendar: Now calendar title-tooltip can show more than 1 event per day.

Version 1.2.6 Maintenance release by svasti
* Bug fixed: number of table columns now determined dynamically
* Bug fixed: switching off field doesn't delete values any more (added all switched off fields as hidden values)
* Beauty fix: Date of multi-day events is presented in a nicer way
* User demand: Date can again be entered directly (date check added, "-,--,?,??" possible as days)
* User demand: Swichted off fields are now also off in all input tables
  (was a big job!) and remailing fields now fill available space automatically
* Some changes in css
* Bug fixed: Enddate in Birthday would screw up the coloring of the calendar display, Now enddate automatically deleted in birthday case
* changed all <input.. and <img .. to tag('input... and tag('img.. so that even the backand validates (could be used on member pages)
* added filepath_data in config
* added possibility to have the notice "no next event sceduled" in next event

Version 1.2 Build 5 by svasti
* Possibility to enter any text in link-text field,
* further cleaning up the code,
* width definition of input fields moved to css, now IE and FF display the input table in the same way,
* sorting on save, function written by manu

Version 1.2 Build 4 by svasti
* added sorting of events in input table on saving,
  suggested by manu in the CMSimple forum.  	  	

Version 1.2 Build 3 by svasti
* added css classes for width of input fields,
  deleated width fields in config. This was necessary
  to give same appearance of input table in IE as in Firefox.

Version 1.2 Build 2 by svasti
* code cleanup in index.php
* Danish language files by maeg

Version 1.2 modified by Holger http://cmsimple.holgerirmler.de - released 04/2011
* Fixed some PHP 5.3.0 deprecated code (split() ...)
* Added xh_debugmode() and a bit security
* Removed direct edit of the eventfiles in admin-area
* Bugfix: layout crashed on eventspage when a birthday (###) was the first event in a month
* Added the DatePicker v5.4 by frequency-decoder.com and made the date inputs readonly
* Replaced ";" by " " in users input to prevent a script-crash and to keep the field-structure in the event-files
* nextevent() will return now nothing when there is no future event
* $plugin_cf['calendar']['date_delimiter']: is ready to use full-stop ".", forward slash "/" and minus/dash "-",
  a changed delimitter at runtime will be converted by the DatePicker, so no crashed eventfiles anymore
* Slightly changed the "narrow" style in the backend

Version 1.1 modified by svasti http://svasti.de - 03/2011
* xh-unicode-compatible
* lots of obsolete attributes moved from config to css (otherwise no validation in html5)
* elimination of unnecessary attributes
* changed obsolete php code ("eregi" is obsolete in php 5.3)
* improvement in backend, 3 input styles available (wide, medium narrow)
* tooltips in config
* extra line in event display giving the period for which the events are listed,
* possibility to show past events, new way to show end of events
* bug corrections: begin and end display of events lasting several days were mixed up
* events from different years would mess up the event-display,
* new way to show end of events (date and/or time),
* correction of other minor bugs (like birthday in calendar view) plus beautifying the code display a bit... but not too much

Version 1.0 By Tory
* The function of showing/not showing time, location and link in event view will now work again

Modified by Bob (cmsimple.fr), 02/2008. (this is from CMSimple Flex package)

Version 0.9 By Tory
* Minor bugs corrected regarding Date-delimiter problem in events in Calendar view.
* Event Page is now configured in the language file. You therefore don't need to add Eventpage-name in Calender call in Template or Calendar-page
* Singular-Plural text regarding age in birtdays in EventPage are now added to language files.
* Parameters added to language files are:
  * $plugin_tx['calendar']['event_page']="Events";  - The Eventpage, in wich Events in Calendar are related to.
  * $plugin_tx['calendar']['age_singular_text']="year"; age text for ages <2 Years
  * $plugin_tx['calendar']['age_plural1_text']="years";
  * age text for ages >1 and <5 years
  * $plugin_tx['calendar']['age_plural2_text']="years"; age text for ages >4 years, used in many east european contries

Version 0.8 By Tory
* Events can now be added / edited / deleted from a CMSimple page, without the Administrator Rights.
  Just create a page and add the following line:
  #CMSimple $output.=EditEvents();#
  The page can be behind Memberspage or Register plugin to allow just a single group of members to be Editor of the Event File. Eg.: #CMSimple $output.=access('admin');$output.=EditEvents();# (Used in combination with register plugin.)
* Only index.php file in Calender plugin has been changed since Version 0.7

Version 0.7 By Tory based on Svarrers version 0.5 : 20. Sep. 2007
* Several dates events are now handled so that they can last over several days, weeks, month or years.
* Begin and End date/time are entered in the eventfile separately. They are only shown in Eventpage with
  the Begin date and End date. In the Calender it is possible to select whether the between dates should be
  higlighted or not.
* In Nextevent and Event-page are automatically written Begin date-time or End date-time to a several dates event.

Version 0.6 Mod by Tory : 16. Sep 2007
* New Function added: nextevent() will display next coming event compared to current date-time.
* New way of sorting events that makes it possible to have more events the same day, and have them sorted

admin.php
<?php
/*
CMSimple - Calendar Plugin - 1.2.10 by svasti 03/2013, version 1.2 modified by Holger, version 1.1 by svasti 3/2011, version 0.9 Mod by Tory 15-01-2008 - Modified by Bob (cmsimple.fr), 02/2008.
*/

if ((!function_exists('sv')) || (strlen($sl) !== 2) || preg_match('/admin.php/i', sv('PHP_SELF')))die('Access denied');

if($calendar){

    $eventfile = $pth['folder']['plugins'].$plugin."/content/eventcalendar_".$sl.".txt";
    if(!is_file($eventfile)){$handle = fopen($eventfile, "w");
    fclose($handle);}

    $o.=print_plugin_admin('on');

    $calendar_credits = "<h2>Calendar plugin version 1.2.10 </h2>\n"
                      . '<p>1.2.10: Maintenance release 03/2013 fixing magic quotes problems, suggested by <a href="http://3-magi.net" target="_blank">cmb</a>' . tag('br')
                      . '1.2.9: Maintenance release 05/2012, on suggestion of <a href="http://3-magi.net" target="_blank">cmb</a>' . tag('br')
                      . '1.2.8: Security release 03/2012, on suggestion of  <a href="http://3-magi.net" target="_blank">cmb</a>' . tag('br')
                      . '1.2.1 - 1.2.7 (10/2011): Maintenance releases  by <a href="http://svasti.de" target="_blank">svasti</a>' . tag('br')
                      . '1.2 with functionality and security added by <a href="http://cmsimple.holgerirmler.de" target="_blank">Holger</a>' . tag('br')
                      . '1.1 (03/2011) with major changes by <a href="http://svasti.de" target="_blank">svasti</a>' . tag('br')
                      . '0.6 - 1.0 by Tory  (and in between mod 02/2008 by Bob (cmsimple.fr))' . tag('br')
                      . '0.1 - 0.5 (2005-2006) by Michael Svarrer</p>';

    if($admin<>'plugin_main'){$o.=plugin_admin_common($action,$admin,$plugin);}

    if($admin=='') {
        $o .= $calendar_credits;
    }

    if($admin=='plugin_main'){
    	$o.=$plugin_tx['calendar']['admin_text_start'];
        $o.=EditEvents($plugin_cf['calendar']['event-input_backend_narrow_medium_or_wide']);
    }
}
?>
